students/DrewSmith/lessons/lesson09/assignment/charges_calc.py

```diff
@@ -0,0 +1,185 @@
+            logging.warn(key_log.format(key, key_err.args[0], str(key_err)))
+            print(f"WARNING: {key} missing {key_err.args[0]} value")
+        except ValueError as value_error:
+            logging.warn(f"Invalid data: {key}: {str(value_error)}")
+            print(f"WARNING: {key} has an invalid rental date value")
+
+        value['total_days'] = (rental_end - rental_start).days
+
+        if value['total_days'] < 0:
+            logging.error(f"Invalid data: {key} rental start " + \
+                f"{rental_start} is greater than the end {rental_end}")
+            print(f"ERROR: {key} rental start is greater than end")
+
+        try:
+            value['total_price'] = value['total_days'] * value['price_per_day']
+        except KeyError as key_err:
+            logging.warn(key_log.format(key, key_err.args[0], str(key_err)))
+            print(f"WARNING: {key} missing {key_err.args[0]} value")
+        except ValueError as value_error:
+            logging.warn(f"Invalid data: {key}: {str(value_error)}")
+            print(f"WARNING: {key} has invalid price_per_day")
+
+        try:
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+        except ValueError as value_error:
+            logging.warn(f"Invalid data: {key}: {str(value_error)}")
+            print(
+                f"WARNING: {key} unable square root the negative total_price")           
+        
+        try:
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+        except KeyError as key_err:
+            logging.warn(key_log.format(key, key_err.args[0], str(key_err)))
+            print(f"WARNING: {key} missing {key_err.args[0]} value")
+        except ValueError as value_error:
+            logging.warn(f"Invalid data: {key}: {str(value_error)}")
+            print(f"WARNING: {key} has invalid units_rented value")
+        except ZeroDivisionError as zero_error:
+            logging.warn(
+                f"Invalid data: {key}: units_rented is 0: {str(zero_error)}")
+            print(f"WARNING: {key} units_rented cannot be 0")
+
+def save_to_json(filename, file_data):
+    '''
+    Outputs the results to the JSON ouput file
+
+    :param filename: output file name
+    :param data: data to write to file
+    '''
+    with open(filename, 'w') as file:
+        json.dump(file_data, file, indent=4)
+        logging.debug(f"Output file generated: {filename}")
+        print("Output file complete")
+
+def parse_cmd_arguments():
+    ''' Processes commandline arguments from the user '''
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='ouput JSON file', required=False)
+
+    parser.add_argument('-l', '--log_off', help='ouput JSON file', required=False)
```

Comment:
Why are the help for all of your outputs the same? You would definitely want to include help for log_off, since it is unusual.